dashboard/test/controllers/api/v1/sections_controller_test.rb

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+require 'test_helper'
+
+class Api::V1::SectionsControllerTest < ActionController::TestCase
+  self.use_transactional_test_case = true
+
+  setup_all do
+    @teacher = create(:teacher)
+
+    @word_section = create(:section, user: @teacher, login_type: 'word')
+    @word_user_1 = create(:follower, section: @word_section).student_user
+
+    @picture_section = create(:section, user: @teacher, login_type: 'picture')
+    @picture_user_1 = create(:follower, section: @picture_section).student_user
+
+    @regular_section = create(:section, user: @teacher, login_type: 'email')
+
+    @flappy_section = create(:section, user: @teacher, login_type: 'word', script_id: Script.get_from_cache(Script::FLAPPY_NAME).id)
+    @flappy_user_1 = create(:follower, section: @flappy_section).student_user
+  end
+
+  setup do
+    # place in setup instead of setup_all otherwise course ends up being serialized
+    # to a file if levelbuilder_mode is true
+    @course = create(:course)
+    @section_with_course = create(:section, user: @teacher, login_type: 'word', course_id: @course.id)
+    @section_with_course_user_1 = create(:follower, section: @section_with_course).student_user
+  end
+
+  test 'returns all sections belonging to teacher' do
+    sign_in @teacher
+
+    get :index
+    assert_response :success
+    json = JSON.parse(@response.body)
+
+    expected = @teacher.sections.map {|section| section.summarize.with_indifferent_access}
+    assert_equal expected, json
+  end
+
+  test 'students own zero sections' do
+    sign_in @word_user_1
+
+    get :index
+    assert_response :success
+    assert_equal '[]', @response.body
```

Comment:
Not that I know of; this is here to preserve existing behavior.  I think it would also be reasonable to return unauthorized here.